python/cudf/cudf/pylibcudf_tests/test_binaryops.py

```diff
@@ -0,0 +1,741 @@
+        binop_out_ty,
+        plc.binaryop.BinaryOperator.NOT_EQUAL,
+    )
+
+
+@pytest.fixture(scope="module")
+def less_tests(binop_lhs_ty, binop_rhs_ty, binop_out_ty):
+    return make_test(
+        binop_lhs_ty,
+        binop_rhs_ty,
+        binop_out_ty,
+        plc.binaryop.BinaryOperator.LESS,
+    )
+
+
+@pytest.fixture(scope="module")
+def greater_tests(binop_lhs_ty, binop_rhs_ty, binop_out_ty):
+    return make_test(
+        binop_lhs_ty,
+        binop_rhs_ty,
+        binop_out_ty,
+        plc.binaryop.BinaryOperator.GREATER,
+    )
+
+
+@pytest.fixture(scope="module")
+def less_equal_tests(binop_lhs_ty, binop_rhs_ty, binop_out_ty):
+    return make_test(
+        binop_lhs_ty,
+        binop_rhs_ty,
+        binop_out_ty,
+        plc.binaryop.BinaryOperator.LESS_EQUAL,
+    )
+
+
+@pytest.fixture(scope="module")
+def greater_equal_tests(binop_lhs_ty, binop_rhs_ty, binop_out_ty):
+    return make_test(
+        binop_lhs_ty,
+        binop_rhs_ty,
+        binop_out_ty,
+        plc.binaryop.BinaryOperator.GREATER_EQUAL,
+    )
+
+
+@pytest.fixture(scope="module")
+def null_equals_tests(binop_lhs_ty, binop_rhs_ty, binop_out_ty):
+    return make_test(
+        binop_lhs_ty,
+        binop_rhs_ty,
+        binop_out_ty,
+        plc.binaryop.BinaryOperator.NULL_EQUALS,
+    )
+
+
+@pytest.fixture(scope="module")
+def null_max_tests(binop_lhs_ty, binop_rhs_ty, binop_out_ty):
+    return make_test(
+        binop_lhs_ty,
+        binop_rhs_ty,
+        binop_out_ty,
+        plc.binaryop.BinaryOperator.NULL_MAX,
+    )
+
+
+@pytest.fixture(scope="module")
+def null_min_tests(binop_lhs_ty, binop_rhs_ty, binop_out_ty):
+    return make_test(
+        binop_lhs_ty,
+        binop_rhs_ty,
+        binop_out_ty,
+        plc.binaryop.BinaryOperator.NULL_MIN,
+    )
+
+
+@pytest.fixture(scope="module")
+def generic_binary_tests(binop_lhs_ty, binop_rhs_ty, binop_out_ty):
+    return make_test(
+        binop_lhs_ty,
+        binop_rhs_ty,
+        binop_out_ty,
+        plc.binaryop.BinaryOperator.GENERIC_BINARY,
+    )
+
+
+@pytest.fixture(scope="module")
+def null_logical_and_tests(binop_lhs_ty, binop_rhs_ty, binop_out_ty):
+    return make_test(
+        binop_lhs_ty,
+        binop_rhs_ty,
+        binop_out_ty,
+        plc.binaryop.BinaryOperator.NULL_LOGICAL_AND,
+    )
+
+
+@pytest.fixture(scope="module")
+def null_logical_or_tests(binop_lhs_ty, binop_rhs_ty, binop_out_ty):
+    return make_test(
+        binop_lhs_ty,
+        binop_rhs_ty,
+        binop_out_ty,
+        plc.binaryop.BinaryOperator.NULL_LOGICAL_OR,
+    )
+
+
+@pytest.fixture(scope="module")
+def invalid_binary_tests(binop_lhs_ty, binop_rhs_ty, binop_out_ty):
+    return make_test(
+        binop_lhs_ty,
+        binop_rhs_ty,
+        binop_out_ty,
+        plc.binaryop.BinaryOperator.INVALID_BINARY,
+    )
+
+
+def test_add(add_tests, columns):
+    _test_binaryop_inner(
+        add_tests, columns, np.add, plc.binaryop.BinaryOperator.ADD
```

Comment:
Rather than using numpy operators, can we pass around `pyarrow.compute.*` functions directly?